nav2_behavior_tree/include/nav2_behavior_tree/plugins/action/truncate_path_local_action.hpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2018 Intel Corporation
+// Copyright (c) 2020 Francisco Martin Rico
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef NAV2_BEHAVIOR_TREE__PLUGINS__ACTION__TRUNCATE_PATH_LOCAL_ACTION_HPP_
+#define NAV2_BEHAVIOR_TREE__PLUGINS__ACTION__TRUNCATE_PATH_LOCAL_ACTION_HPP_
+
+#include <memory>
+#include <string>
+
+#include "nav_msgs/msg/path.hpp"
+
+#include "behaviortree_cpp_v3/action_node.h"
+#include "tf2_ros/buffer.h"
+
+namespace nav2_behavior_tree
+{
+
+/**
+ * @brief A BT::ActionNodeBase to shorten path by some distance
+ */
+class TruncatePathLocal : public BT::ActionNodeBase
```

Comment:
Add the new node to the list https://github.com/ros-planning/navigation2/blob/main/nav2_behavior_tree/nav2_tree_nodes.xml so that users can use Groot with it